tools/codegen/gen.py

```diff
@@ -474,7 +474,18 @@ def go(f: NativeFunction) -> Optional[str]:
 
         dispatcher_returns_type = dispatcher.returns_type(f.func.returns)
         dispatcher_args = dispatcher.arguments(f.func)
-        dispatcher_exprs = dispatcher.legacydispatcherarguments_exprs(legacy_dispatcher_args)
+
+
+        if local.use_c10_dispatcher() is UseC10Dispatcher.full:
```

Comment:
I'm wondering if you can just get rid of the `local.use_c10_dispatcher() is UseC10Dispatcher.full` casing here entirely.  Right now, in BackendSelect you are using this to control whether or not you apply `hacky_wrapper_for_legacy_signatures`. But you are 100% in charge of *generating* the BackendSelect functions in the first place. So if you just generate the BackendSelect function to have the non-legacy signature in the first place, then you don't need any of this business at all.

Reply:
That's basically what I'm doing with this PR, I'm generating them with the non-legacy signature directly instead of wrapping it. But I can only do that for c10-full ops since the concept of a "non-legacy signature" only exists for c10-full ops. Non-c10-full ops always use the legacy signature in their unboxed calls. That's why I still need the casing here.